AudioManagerUtilities/src/CAmSocketHandler.cpp

```diff
@@ -559,23 +569,31 @@ am_Error_e CAmSocketHandler::addTimer(const timespec & timeouts, std::function<v
         return err;
     }
 
-    static auto actionPoll = [](const pollfd pollfd, const sh_pollHandle_t handle, void* userData)
+    auto actionPoll = [this](const pollfd pollfd, const sh_pollHandle_t handle, void* userData)
     {
-        uint64_t mExpirations;
-        if (read(pollfd.fd, &mExpirations, sizeof(uint64_t)) == -1)
-        {
-            //error received...try again
-            read(pollfd.fd, &mExpirations, sizeof(uint64_t));
-        }
+        uint64_t expCnt;
+        ssize_t bytes = read(pollfd.fd, &expCnt, sizeof(expCnt));
+        if (bytes == sizeof(expCnt))
+            return;
+
+        // ppoll has to be called again in following case
+        if ((bytes == -1) && (errno == EAGAIN))
+            return;
+
+        // failed to read data from timer_fd...
+        std::ostringstream msg;
+        msg << "Failed to read from timer fd: " << pollfd.fd << " errno: " << std::strerror(errno);
+        throw std::runtime_error(msg.str());
     };
 
-    err = addFDPoll(timerItem.fd, POLLIN, NULL, actionPoll, [callback](const sh_pollHandle_t handle, void* userData)->bool
-    {
-        callback(handle, userData);
-        return false;
-    },
-    NULL, userData, handle);
-    if (E_OK == err)
+    err = addFDPoll(timerItem.fd, POLLIN | POLLERR, NULL, actionPoll,
+                        [callback](const sh_pollHandle_t handle, void* userData)->bool {
```

Comment:
(Style)
In this code I never figured out why one lambda is assigned to a variable above (line 572 in the new version), and the others are done inline.  Seems like a nice and consistent way would be preferable here.
--> OK (for now)